app/src/main/res/layout/activity_register.xml

```diff
@@ -1,217 +1,220 @@
+            app:met_textColorHint="#bdbdbd"
 
             />
+
         <com.rengwuxian.materialedittext.MaterialEditText
+            android:id="@+id/et_phone"
             android:layout_width="match_parent"
             android:layout_height="0dp"
-            android:layout_weight="1"
-            android:hint="phone"
+            android:layout_marginRight="20dp"
 
-            android:id="@+id/et_phone"
             android:layout_marginStart="10dp"
             android:layout_marginTop="10dp"
-            android:layout_marginRight="20dp"
-            app:met_textColorHint="#bdbdbd"
+            android:layout_weight="1"
+            android:hint="phone"
+            android:inputType="phone"
 
 
+            android:maxLines="1"
+            android:textSize="15dp"
             app:met_baseColor="#ffff00"
-            app:met_primaryColor="#76ff03"
-            app:met_errorColor="#ddaa00"
 
-            android:textSize="15dp"
+            app:met_errorColor="#ddaa00"
 
 
             app:met_floatingLabel="highlight"
             app:met_floatingLabelText="Phone"
             app:met_floatingLabelTextSize="10dp"
-            android:maxLines="1"
-            android:inputType="phone"
+            app:met_primaryColor="#76ff03"
+            app:met_textColorHint="#bdbdbd"
 
             />
+
         <com.rengwuxian.materialedittext.MaterialEditText
+            android:id="@+id/et_email"
             android:layout_width="match_parent"
             android:layout_height="0dp"
-            android:layout_weight="1"
-            android:hint="email"
+            android:layout_marginRight="20dp"
 
-            android:id="@+id/et_email"
             android:layout_marginStart="10dp"
             android:layout_marginTop="10dp"
-            android:layout_marginRight="20dp"
-            app:met_textColorHint="#bdbdbd"
+            android:layout_weight="1"
+            android:hint="email"
+            android:inputType="textEmailAddress"
 
 
+            android:maxLines="1"
+            android:textSize="15dp"
             app:met_baseColor="#ffff00"
-            app:met_primaryColor="#76ff03"
-            app:met_errorColor="#ddaa00"
 
-            android:textSize="15dp"
+            app:met_errorColor="#ddaa00"
 
 
             app:met_floatingLabel="highlight"
             app:met_floatingLabelText="Email"
             app:met_floatingLabelTextSize="10dp"
-            android:maxLines="1"
-            android:inputType="text"
+            app:met_primaryColor="#76ff03"
+            app:met_textColorHint="#bdbdbd"
 
             />
 
         <RadioGroup
+            android:id="@+id/hostel"
             android:layout_width="match_parent"
             android:layout_height="0dp"
             android:layout_weight="1"
-            android:orientation="horizontal"
-            android:id="@+id/hostel">
+            android:orientation="horizontal">
 
 
             <RadioButton
+                android:id="@+id/hosteller"
```

Comment:
id should be like rb_hosteller, hosteller does not imply that this is a radioButton